cpp/include/cuml/decomposition/sign_flip_mg.hpp

```diff
@@ -40,17 +40,17 @@ void sign_flip(raft::handle_t& handle,
                std::vector<MLCommon::Matrix::Data<float>*>& input_data,
                MLCommon::Matrix::PartDescriptor& input_desc,
                float* components,
-               int n_components,
+               std::size_t n_components,
                cudaStream_t* streams,
-               int n_stream);
+               std::size_t n_stream);
 
 void sign_flip(raft::handle_t& handle,
                std::vector<MLCommon::Matrix::Data<double>*>& input_data,
                MLCommon::Matrix::PartDescriptor& input_desc,
                double* components,
-               int n_components,
+               std::size_t n_components,
                cudaStream_t* streams,
-               int n_stream);
+               std::size_t n_stream);
```

Comment:
It's not likely these values are ever going to need to be 64-bit. I'd prefer to use `std::uint32_t` for this so we aren't automatically promoting everything to 64-bit.